src/imageengine.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import Plugin from '../core/plugin.js';
+import buildModelConverter from '../engine/conversion/buildmodelconverter.js';
+import WidgetEngine from './widget/widgetengine.js';
+import { modelToViewImage, viewToModelImage, modelToViewSelection } from './converters.js';
+import { toImageWidget } from './utils.js';
+
+/**
+ * The image engine plugin.
+ * Registers `image` as a block element in document's schema and allows it to have two attributes: `src` and `alt`.
+ * Registers converters for editing and data pipelines.
+ *
+ * @memberof image
+ * @extends core.Plugin.
+ */
+export default class ImageEngine extends Plugin {
+	/**
+	 * @inheritDoc
+	 */
+	static get requires() {
+		return [ WidgetEngine ];
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	init() {
+		const editor = this.editor;
+		const document = editor.document;
+		const dataPipeline = editor.data;
+		const editingPipeline = editor.editing;
+
+		// Configure schema.
+		document.schema.registerItem( 'image', '$block' );
```

Comment:
Additionally, `'image'` should be added to `schema.objects` set. It's used by the `insertContent` algorithm to make some decisions on how to selected the pasted thing. And it can be used by more algorithms (e.g. the keyboard support).